subsys/net/lib/lwm2m/lwm2m_registry.c

```diff
@@ -589,6 +601,33 @@ static int lwm2m_engine_set(const char *pathstr, const void *value, uint16_t len
 		*((uint8_t *)data_ptr) = *(uint8_t *)value;
 		break;
 
+	case LWM2M_RES_TYPE_TIME:
+		if (!lwm2m_validate_time_resource_lenghts(max_data_len, len)) {
+			LOG_ERR("Time Set: buffer length %u  max data len %u not supported", len,
+				max_data_len);
+			return -EINVAL;
+		}
+
+		if (max_data_len == sizeof(time_t)) {
+			if (len == sizeof(time_t)) {
+				*((time_t *)data_ptr) = *(time_t *)value;
+			} else {
+				LOG_WRN("Time Set by 32-bit to 64-bit resource:%s", pathstr);
+				*((time_t *)data_ptr) = (time_t) *((uint32_t *)value);
+				break;
+			}
+		} else {
+			LOG_WRN("Time Set to 32-bit resource size may overflow resource:%s",
```

Comment:
I wonder, do we even need to support this case? Can't we enforce that the application uses `time_t` as a storage for the resource? Just like with float - the server may use either 4 or 8 byte representation, but on the application side we only allow to use `double` as a storage.

Reply:
I added this because Application side may still use smaller 32-bit buffer. Testing with nrf-sdk I see this print coming some object for store timestamp to smaller than time_t.

Reply:
That was my point, I think the library could require that the application uses `time_t` to represent time (just as we require to use `double` to represent float) instead of covering corner cases within the library. But no strong push here, we can do it this way as well.